collector/load_balancer_collector_test.go

```diff
@@ -104,120 +104,267 @@ func TestLoadBalancerCollector_GenerateLoadBalancerStatusMetrics(t *testing.T) {
 			},
 			expectedMetrics: []loadBalancerStatusMetric{
 				{
-					ID:     "fake-load-balancer-id-01",
-					Name:   "fake-load-balancer-name-01",
-					Status: 1.0,
+					ID:   "fake-load-balancer-id-01",
+					Name: "fake-load-balancer-name-01",
+					StatusDetail: map[string]float64{
```

Comment:
the maps seem to be repeated in several places, can this be var/const instead defined outside?

Reply:
update: just realized the values are different in each case, how about extracting to function instead?

Reply:
something like `buildStatusDetail("UP", 1.0)` which will return map with default 0.0 for all key-value pairs except `UP`

Reply:
The logic to construct status detail is the same everywhere, wdyt if constructStatusDetail detached from loadBalancerCollector? Then we can use this function on the test and other metric

Reply:
or can accept slice of maps as well only for non-zero status
```
type nonZeroStates map[string]float64
...
buildStatusDetails(nonZeroStates{"UP": 1.0})
```

Reply:
Ah, that also works.
Will implement this